aws/resource_aws_datapipeline_test.go

```diff
@@ -0,0 +1,269 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSDataPipelineExists(resourceName, &conf),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
+					resource.TestCheckResourceAttr(
+						resourceName, "tags.foo", "bar"),
+					resource.TestCheckResourceAttr(
+						resourceName, "tags.fizz", "buzz"),
+				),
+			},
+			{
+				Config: testAccAWSDataPipelineConfigWithTags(rName, "foo", "bar2", "fizz2", "buzz2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSDataPipelineExists(resourceName, &conf),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
+					resource.TestCheckResourceAttr(
+						resourceName, "tags.foo", "bar2"),
+					resource.TestCheckResourceAttr(
+						resourceName, "tags.fizz2", "buzz2"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAWSDataPipelineConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSDataPipelineExists(resourceName, &conf),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "0"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSDataPipelineDisappears(conf *datapipeline.PipelineDescription) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).datapipelineconn
+		params := &datapipeline.DeletePipelineInput{
+			PipelineId: conf.PipelineId,
+		}
+
+		_, err := conn.DeletePipeline(params)
+		if err != nil {
+			return err
+		}
+		return waitForDataPipelineDeletion(conn, *conf.PipelineId)
+	}
+}
+
+func testAccCheckAWSDataPipelineDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).datapipelineconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_datapipeline" {
+			continue
+		}
+		// Try to find the Pipeline
+		var err error
+		resp, err := conn.DescribePipelines(
+			&datapipeline.DescribePipelinesInput{
+				PipelineIds: []*string{aws.String(rs.Primary.ID)},
+			})
+
+		if err != nil {
+			if isAWSErr(err, datapipeline.ErrCodePipelineNotFoundException, "") {
+				continue
+			} else if isAWSErr(err, datapipeline.ErrCodePipelineDeletedException, "") {
+				continue
+			}
+			return err
+		}
+		if len(resp.PipelineDescriptionList) != 0 &&
+			*resp.PipelineDescriptionList[0].PipelineId == rs.Primary.ID {
+			return fmt.Errorf("Pipeline still exists")
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAWSDataPipelineExists(n string, v *datapipeline.PipelineDescription) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No DataPipeline ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).datapipelineconn
+
+		opts := &datapipeline.DescribePipelinesInput{
```

Comment:
Same here. 👍 It seems like this logic should be replaced with `resourceAwsDataPipelineRetrieve()`

Reply:
fixed